### DISCUSSION TITLE
null pointer in oidc_util_jwt_create()

### DISCUSSION THREAD

**Discussion by marcstern** [OpenIDC/mod_auth_openidc#1030] [General]
I got a null pointer exception and I found the cause.
On line 135 of util.c, we have the line
```
if (oidc_util_create_symmetric_key(r, secret, 0, OIDC_JOSE_ALG_SHA256, FALSE, &jwk) == FALSE)
		goto end;
```
In case secret is null (I guess it happens when OIDCClientSecret is not set), oidc_util_create_symmetric_key() returns TRUE but jwk is still null. Afew lines later, we call oidc_jwt_sign() where that generates an exception (jwk->cjose_jwk).

A check for nullity should definitely be added, but I'm not sure if the best place is oidc_util_jwt_create() or in a calling function. Any idea?
In case of doubt, we could begin to place it here until we find a better solution ;-)

**Comment by zandbelt** [comment]
thanks for reporting: I believe this could only happen if `OIDCCryptoPassphrase` is not set and a check for that should happen at startup, at least for those use cases where it is required ([OIDC](https://github.com/OpenIDC/mod_auth_openidc/blob/v2.4.12.3/src/config.c#L2483-L2484), or [OAuth 2.0 RS with encrypted cache](https://github.com/OpenIDC/mod_auth_openidc/blob/v2.4.12.3/src/config.c#L2577-L2578)). So please share your config to find out where the fix needs to go.

**Comment by marcstern** [comment]
Correct: missing OIDCCryptoPassphrase is the culprit

**Reply by zandbelt** [reply]
I'm still interested in your config as the config checks at startup should have caught this (so they need a fix at least)

**Comment by marcstern** [comment]
Difficult as it encompasses a lot of macros, but I think any working config should generate the problem if you remove the OIDCCryptoPassphrase directive

**Reply by zandbelt** [reply]
no, that's what the checks are for...

**Reply by zandbelt** [reply]
what AuthType and cache options have you set?

**Reply by zandbelt** [reply]
also, a full server debug log of the calling stack would be helpful to determine in what setting this happens

**Comment by marcstern** [comment]
I'll try to prepare that, but I probably won't be able to do it before 2 weeks.

**Reply by zandbelt** [reply]
thanks for your help
